api/v1alpha3/types.go

```diff
@@ -278,16 +278,25 @@ type AvailabilityZone struct {
 }
 
 // VMIdentity defines the identity of the virtual machine, if configured.
-// +kubebuilder:validation:Enum=None;SystemAssigned
+// +kubebuilder:validation:Enum=None;SystemAssigned;UserAssigned
 type VMIdentity string
 
 const (
 	// VMIdentityNone ...
 	VMIdentityNone VMIdentity = "None"
 	// VMIdentitySystemAssigned ...
 	VMIdentitySystemAssigned VMIdentity = "SystemAssigned"
+	// VMIdentityUserAssigned ...
+	VMIdentityUserAssigned VMIdentity = "UserAssigned"
 )
 
+// UserAssignedIdentity defines the user-assigned identities provided
+// by the user to be assigned to Azure resources.
+type UserAssignedIdentity struct {
+	// ProviderID is the identification ID of the user-assigned Identity
+	ProviderID string `json:"providerID,omitempty"`
```

Comment:
This should be required, not optional.

Reply:
Just to be clear, this is only required for now as we only support BYO identity correct?

Reply:
Precisely. We can make it optional in the future and it'll be non-breaking.

Reply:
done